zowe/explorer-api-common
# SSO implementation - Support for use of apimlAuthenticationToken (#43)

* extract apimlAuthorizationToken and pass through for all auth

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* fix linting errors

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* fix boolean check

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* add back ApiControllerTest and fix exception construction

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* logon to api gateway

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* switch to constructor for intialising logon

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* Get token when provided in auth header too

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* ensure cookie header is not null before tring to get token from it

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* debug getting apimlauthtoken

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* check for null auth header before checking if empty

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* add getter for gateway port

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* Cleanup unused debug statements

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* Remove logout matcher and revoke unneccessary copyright update

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* Missed copyright update

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>

* version bump

Signed-off-by: Jordan Cain <jordan.cain1@uk.ibm.com>
